### PR TITLE
feat(manager/npm): update yarn metadata versions

### LINKED ISSUE #9516: Yarn 2 - dependenciesMeta aren't updated
**What Renovate type, platform and version are you using?**

Hosted Renovate over GitHub app

**Describe the bug**

Yarn 2 allows for users to list some information about dependencies like the following:
```
  "dependenciesMeta": {
    "lodash@4.16.0": {
      "unplugged": true
    }
  }
```

when Renovate tries to update lodash, it should also update this entry, otherwise the mismatch will cause confusing errors. Currently, Renovate just ignores it entirely.

also note that there are other "meta" fields -- see https://yarnpkg.com/configuration/manifest/#dependenciesMeta

**Relevant debug logs**

```
INFO: Repository started
{
  "renovateVersion": "24.113.0"
}
DEBUG: Using localDir: /mnt/renovate/gh/AaronBuxbaum/renovate-repro
DEBUG: Repository cache is valid
DEBUG: initRepo("AaronBuxbaum/renovate-repro")
DEBUG: Overriding default GitHub endpoint
{
  "endpoint": "https://api.github.com/"
}
DEBUG: No dangling containers to remove
DEBUG: AaronBuxbaum/renovate-repro default branch = master
DEBUG: Using app token for git init
DEBUG: resetMemCache()
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findFile(renovate.json)
DEBUG: Initializing git repository into /mnt/renovate/gh/AaronBuxbaum/renovate-repro
DEBUG: git clone completed
{
  "durationMs": 784
}
DEBUG: latest repository commit
{
  "latestCommit": {
    "hash": "c5db84984f287dce15a447655a1046a6fdd5bb2c",
    "date": "2021-04-12T13:17:34-04:00",
    "message": "Make it a little more complex",
    "refs": "HEAD -> master, origin/master, origin/HEAD",
    "body": "",
    "author_name": "Aaron Buxbaum",
    "author_email": "me@aaronbuxbaum.com"
  }
}
DEBUG: Setting git author name
{
  "gitAuthorName": "Renovate Bot"
}
DEBUG: Setting git author email
{
  "gitAuthorEmail": "bot@renovateapp.com"
}
DEBUG: Config file exists
{
  "fileName": "renovate.json"
}
DEBUG: Retrieving issueList
DEBUG: Retrieved 0 issues
DEBUG: Repo is onboarded
DEBUG: Found renovate.json config file
DEBUG: Repository config
{
  "fileName": "renovate.json",
  "config": {
    "extends": [
      "config:base"
    ]
  }
}
DEBUG: migrateAndValidate()
DEBUG: No config migration necessary
DEBUG: massaged config
{
  "config": {
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "config:base"
    ]
  }
}
DEBUG: migrated config
{
  "config": {
    "extends": [
      "github>whitesource/merge-confidence:beta",
      "config:base"
    ]
  }
}
DEBUG: Found repo ignorePaths
{
  "ignorePaths": [
    "**/node_modules/**",
    "**/bower_components/**",
    "**/vendor/**",
    "**/examples/**",
    "**/__tests__/**",
    "**/test/**",
    "**/tests/**",
    "**/__fixtures__/**"
  ]
}
DEBUG: detectSemanticCommits()
DEBUG: getCommitMessages
DEBUG: Semantic commits detection: unknown
DEBUG: No semantic commits detected
DEBUG: Setting branchPrefix: renovate/
DEBUG: No vulnerability alerts found
DEBUG: No vulnerability alerts found
DEBUG: processRepo()
DEBUG: No baseBranches
DEBUG: extract()
DEBUG: Setting current branch to master
DEBUG: latest commit
{
  "branchName": "master",
  "latestCommitDate": "2021-04-12T13:17:34-04:00"
}
DEBUG: Using file match: (^|/)tasks/[^/]+\.ya?ml$ for manager ansible
DEBUG: Using file match: (^|/)requirements\.ya?ml$ for manager ansible-galaxy
DEBUG: Using file match: (^|/)galaxy\.ya?ml$ for manager ansible-galaxy
DEBUG: Using file match: azure.*pipelines?.*\.ya?ml$ for manager azure-pipelines
DEBUG: Using file match: (^|/)batect(-bundle)?\.yml$ for manager batect
DEBUG: Using file match: (^|/)batect$ for manager batect-wrapper
DEBUG: Using file match: (^|/)WORKSPACE(|\.bazel)$ for manager bazel
DEBUG: Using file match: \.bzl$ for manager bazel
DEBUG: Using file match: buildkite\.ya?ml for manager buildkite
DEBUG: Using file match: \.buildkite/.+\.ya?ml$ for manager buildkite
DEBUG: Using file match: (^|/)Gemfile$ for manager bundler
DEBUG: Using file match: (^|/)Cargo.toml$ for manager cargo
DEBUG: Using file match: (^|/).circleci/config.yml$ for manager circleci
DEBUG: Using file match: (^|/)cloudbuild.ya?ml for manager cloudbuild
DEBUG: Using file match: (^|/)Podfile$ for manager cocoapods
DEBUG: Using file match: (^|/)([\w-]*)composer.json$ for manager composer
DEBUG: Using file match: (^|/)deps\.edn$ for manager deps-edn
DEBUG: Using file match: (^|/)docker-compose[^/]*\.ya?ml$ for manager docker-compose
DEBUG: Using file match: (^|/|\.)Dockerfile$ for manager dockerfile
DEBUG: Using file match: (^|/)Dockerfile\.[^/]*$ for manager dockerfile
DEBUG: Using file match: (^|/).drone.yml$ for manager droneci
DEBUG: Using file match: (^|/).gitmodules$ for manager git-submodules
DEBUG: Using file match: ^\.github\/workflows\/[^/]+\.ya?ml$ for manager github-actions
DEBUG: Using file match: \.gitlab-ci\.yml$ for manager gitlabci
DEBUG: Using file match: ^\.gitlab-ci\.yml$ for manager gitlabci-include
DEBUG: Using file match: (^|/)go.mod$ for manager gomod
DEBUG: Using file match: \.gradle(\.kts)?$ for manager gradle
DEBUG: Using file match: (^|/)gradle.properties$ for manager gradle
DEBUG: Using file match: (^|/)gradle.properties$ for manager gradle-lite
DEBUG: Using file match: \.gradle(\.kts)?$ for manager gradle-lite
DEBUG: Using file match: (^|/)gradle/wrapper/gradle-wrapper.properties$ for manager gradle-wrapper
DEBUG: Using file match: (^|/)requirements\.yaml$ for manager helm-requirements
DEBUG: Using file match: (^|/)values.yaml$ for manager helm-values
DEBUG: Using file match: (^|/)helmfile.yaml$ for manager helmfile
DEBUG: Using file match: (^|/)Chart.yaml$ for manager helmv3
DEBUG: Using file match: ^Formula/[^/]+[.]rb$ for manager homebrew
DEBUG: Using file match: \.html?$ for manager html
DEBUG: Using file match: (^|/)plugins\.(txt|ya?ml)$ for manager jenkins
DEBUG: Using file match: (^|/)kustomization\.yaml for manager kustomize
DEBUG: Using file match: (^|/)project\.clj$ for manager leiningen
DEBUG: Using file match: \.pom\.xml$ for manager maven
DEBUG: Using file match: (^|/)pom\.xml$ for manager maven
DEBUG: Using file match: (^|/)package.js$ for manager meteor
DEBUG: Using file match: (^|/)mix\.exs$ for manager mix
DEBUG: Using file match: ^.node-version$ for manager nodenv
DEBUG: Using file match: (^|/)package.json$ for manager npm
DEBUG: Using file match: \.(?:cs|fs|vb)proj$ for manager nuget
DEBUG: Using file match: \.(?:props|targets)$ for manager nuget
DEBUG: Using file match: \.config\/dotnet-tools\.json$ for manager nuget
DEBUG: Using file match: (^|/)\.nvmrc$ for manager nvm
DEBUG: Using file match: (^|/)([\w-]*)requirements.(txt|pip)$ for manager pip_requirements
DEBUG: Using file match: (^|/)setup.py$ for manager pip_setup
DEBUG: Using file match: (^|/)Pipfile$ for manager pipenv
DEBUG: Using file match: (^|/)pyproject\.toml$ for manager poetry
DEBUG: Using file match: (^|/)\.pre-commit-config\.yaml$ for manager pre-commit
DEBUG: Using file match: (^|/)pubspec\.ya?ml$ for manager pub
DEBUG: Using file match: (^|/)\.ruby-version$ for manager ruby-version
DEBUG: Using file match: \.sbt$ for manager sbt
DEBUG: Using file match: project/[^/]*.scala$ for manager sbt
DEBUG: Using file match: (^|/)setup\.cfg$ for manager setup-cfg
DEBUG: Using file match: (^|/)Package\.swift for manager swift
DEBUG: Using file match: \.tf$ for manager terraform
DEBUG: Using file match: (^|/)\.terraform-version$ for manager terraform-version
DEBUG: Using file match: (^|/)terragrunt\.hcl$ for manager terragrunt
DEBUG: Using file match: (^|/)\.terragrunt-version$ for manager terragrunt-version
DEBUG: Using file match: ^.travis.yml$ for manager travis
DEBUG: Matched 1 file(s) for manager npm: package.json
DEBUG: npm file package.json has name "renovate-repro"
DEBUG: Detecting Lerna and Yarn Workspaces
DEBUG: Finding locked versions
DEBUG: Found npm package files
DEBUG: Found 1 package file(s)
INFO: Dependency extraction complete
{
  "baseBranch": "master",
  "stats": {
    "managers": {
      "npm": {
        "fileCount": 1,
        "depCount": 2
      }
    },
    "total": {
      "fileCount": 1,
      "depCount": 2
    }
  }
}
DEBUG: Package releases lookups complete
{
  "baseBranch": "master"
}
DEBUG: branchifyUpgrades
DEBUG: Using group branchName template
DEBUG: Dependency lodash is part of group lodash monorepo
DEBUG: 1 flattened updates found: lodash
DEBUG: Returning 1 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
{
  "config": {
    "npm": [
      {
        "packageFile": "package.json",
        "deps": [
          {
            "depType": "dependencies",
            "depName": "lodash",
            "currentValue": "^4.16.0",
            "datasource": "npm",
            "prettyDepType": "dependency",
            "depIndex": 0,
            "updates": [
              {
                "updateType": "pin",
                "isPin": true,
                "newValue": "4.17.21",
                "newMajor": 4,
                "branchName": "renovate/pin-dependencies"
              }
            ],
            "warnings": [],
            "sourceUrl": "https://github.com/lodash/lodash",
            "homepage": "https://lodash.com/"
          },
          {
            "depType": "resolutions",
            "depName": "lodash",
            "currentValue": "patch:lodash@npm:4.16.0#patches/lodash.patch",
            "skipReason": "unknown-version",
            "prettyDepType": "resolutions",
            "depIndex": 1,
            "updates": []
          }
        ],
        "packageJsonName": "renovate-repro",
        "packageJsonType": "app",
        "yarnLock": "yarn.lock",
        "managerData": {},
        "skipInstalls": true,
        "constraints": {
          "yarn": "^2.2.0"
        },
        "lockFiles": [
          "yarn.lock"
        ]
      }
    ]
  }
}
DEBUG: processRepo()
DEBUG: Processing 1 branch: renovate/pin-dependencies
DEBUG: Calculating hourly PRs remaining
DEBUG: Retrieving PR list
DEBUG: Retrieved 2 Pull Requests
DEBUG: currentHourStart=2021-04-12T17:00:00.000+00:00
DEBUG: PR hourly limit remaining: 1
DEBUG: Calculating prConcurrentLimit (20)
DEBUG: getBranchPr(renovate/pin-dependencies)
DEBUG: findPr(renovate/pin-dependencies, undefined, open)
DEBUG: findPr(renovate/pin-dependencies, undefined, closed)
DEBUG: 0 PRs are currently open
DEBUG: PR concurrent limit remaining: 20
DEBUG: Calculated maximum PRs remaining this run
{
  "prsRemaining": 1
}
DEBUG: PullRequests limit = 1
DEBUG: Calculating branchConcurrentLimit (20)
DEBUG: 0 already existing branches found:
DEBUG: Branch concurrent limit remaining: 20
DEBUG: Calculated maximum branches remaining this run
{
  "branchesRemaining": 20
}
DEBUG: Branches limit = 20
DEBUG: processBranch with 1 upgrades(branch="renovate/pin-dependencies")
DEBUG: Setting current branch to master(branch="renovate/pin-dependencies")
DEBUG: latest commit(branch="renovate/pin-dependencies")
{
  "branchName": "master",
  "latestCommitDate": "2021-04-12T13:17:34-04:00"
}
DEBUG: getBranchPr(renovate/pin-dependencies)(branch="renovate/pin-dependencies")
DEBUG: findPr(renovate/pin-dependencies, undefined, open)(branch="renovate/pin-dependencies")
DEBUG: findPr(renovate/pin-dependencies, undefined, closed)(branch="renovate/pin-dependencies")
DEBUG: branchExists=false(branch="renovate/pin-dependencies")
DEBUG: Branch has 1 upgrade(s)(branch="renovate/pin-dependencies")
DEBUG: recreateClosed is true(branch="renovate/pin-dependencies")
DEBUG: Checking schedule(at any time, null)(branch="renovate/pin-dependencies")
DEBUG: No schedule defined(branch="renovate/pin-dependencies")
DEBUG: Branch needs creating(branch="renovate/pin-dependencies")
DEBUG: Using reuseExistingBranch: false(branch="renovate/pin-dependencies")
DEBUG: manager.getUpdatedPackageFiles() reuseExistinbranch=false(branch="renovate/pin-dependencies")
DEBUG: npm.updateDependency(): dependencies.lodash = 4.17.21(branch="renovate/pin-dependencies")
DEBUG: Upgraded dependency exists in yarn resolutions but is different version(branch="renovate/pin-dependencies")
{
  "depName": "lodash",
  "depKey": "lodash",
  "oldVersion": "^4.16.0",
  "resolutionsVersion": "patch:lodash@npm:4.16.0#patches/lodash.patch"
}
DEBUG: Updating lodash in package.json(branch="renovate/pin-dependencies")
DEBUG: Updated 1 package files(branch="renovate/pin-dependencies")
DEBUG: Getting updated lock files(branch="renovate/pin-dependencies")
DEBUG: Writing package.json files(branch="renovate/pin-dependencies")
{
  "packageFiles": [
    "package.json"
  ]
}
DEBUG: Writing any updated package files(branch="renovate/pin-dependencies")
DEBUG: Writing package.json(branch="renovate/pin-dependencies")
DEBUG: No npmrc file found in repository(branch="renovate/pin-dependencies")
DEBUG: Writing updated .npmrc file to /mnt/renovate/gh/AaronBuxbaum/renovate-repro/.npmrc(branch="renovate/pin-dependencies")
DEBUG: Generating yarn.lock for .(branch="renovate/pin-dependencies")
DEBUG: Spawning yarn install to create /mnt/renovate/gh/AaronBuxbaum/renovate-repro/yarn.lock(branch="renovate/pin-dependencies")
DEBUG: No node constraint found - using latest(branch="renovate/pin-dependencies")
DEBUG: Using docker to execute(branch="renovate/pin-dependencies")
DEBUG: No tag or tagConstraint specified(branch="renovate/pin-dependencies")
{
  "image": "docker.io/renovate/node"
}
DEBUG: Fetching Docker image: docker.io/renovate/node(branch="renovate/pin-dependencies")
DEBUG: Finished fetching Docker image(branch="renovate/pin-dependencies")
DEBUG: Executing command(branch="renovate/pin-dependencies")
{
  "command": "docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/AaronBuxbaum/renovate-repro\":\"/mnt/renovate/gh/AaronBuxbaum/renovate-repro\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_HTTP_TIMEOUT -e YARN_ENABLE_SCRIPTS -w \"/mnt/renovate/gh/AaronBuxbaum/renovate-repro\" docker.io/renovate/node bash -l -c \"npm i -g yarn && yarn install\""
}
DEBUG: exec completed(branch="renovate/pin-dependencies")
{
  "cmd": "docker run --rm --name=renovate_node --label=renovate_child -v \"/mnt/renovate/gh/AaronBuxbaum/renovate-repro\":\"/mnt/renovate/gh/AaronBuxbaum/renovate-repro\" -v \"/tmp/renovate-cache\":\"/tmp/renovate-cache\" -e NPM_CONFIG_CACHE -e npm_config_store -e CI -e YARN_HTTP_TIMEOUT -e YARN_ENABLE_SCRIPTS -w \"/mnt/renovate/gh/AaronBuxbaum/renovate-repro\" docker.io/renovate/node bash -l -c \"npm i -g yarn && yarn install\"",
  "durationMs": 8897,
  "stdout": "\n> yarn@1.22.10 preinstall /home/ubuntu/.npm-global/lib/node_modules/yarn\n> :; (node ./preinstall.js > /dev/null 2>&1 || true)\n\n/home/ubuntu/.npm-global/bin/yarn -> /home/ubuntu/.npm-global/lib/node_modules/yarn/bin/yarn.js\n/home/ubuntu/.npm-global/bin/yarnpkg -> /home/ubuntu/.npm-global/lib/node_modules/yarn/bin/yarn.js\n+ yarn@1.22.10\nadded 1 package in 1.687s\n➤ YN0000: ┌ Resolution step\n➤ YN0000: └ Completed in 0s 300ms\n➤ YN0000: ┌ Fetch step\n➤ YN0013: │ No packages were cached - one package had to be fetched\n➤ YN0019: │ lodash-npm-4.16.0-63abe90291-638e47679f.zip appears to be unused - removing\n➤ YN0019: │ lodash-patch-0b3fe55333-31aafda596.zip appears to be unused - removing\n➤ YN0000: └ Completed in 2s 65ms\n➤ YN0000: ┌ Link step\n➤ YN0000: └ Completed in 0s 308ms\n➤ YN0000: Done in 2s 719ms\n",
  "stderr": ""
}
DEBUG: yarn.lock needs updating(branch="renovate/pin-dependencies")
DEBUG: updateYarnOffline resolvedPaths(branch="renovate/pin-dependencies")
{
  "resolvedPaths": [
    ".yarn/cache",
    ".pnp"
  ]
}
DEBUG: Updated 5 lock files(branch="renovate/pin-dependencies")
{
  "updatedArtifacts": [
    "yarn.lock",
    ".pnp.js",
    ".yarn/cache/lodash-npm-4.17.21-6382451519-4983720b9a.zip",
    ".yarn/cache/lodash-npm-4.16.0-63abe90291-638e47679f.zip (delete)",
    ".yarn/cache/lodash-patch-0b3fe55333-31aafda596.zip (delete)"
  ]
}
DEBUG: 5 file(s) to commit(branch="renovate/pin-dependencies")
DEBUG: Committing files to branch renovate/pin-dependencies(branch="renovate/pin-dependencies")
DEBUG: git commit(branch="renovate/pin-dependencies")
{
  "result": {
    "author": null,
    "branch": "renovate/pin-dependencies",
    "commit": "a946966",
    "root": false,
    "summary": {
      "changes": 6,
      "insertions": 31,
      "deletions": 38
    }
  }
}
DEBUG: git push(branch="renovate/pin-dependencies")
{
  "result": {
    "pushed": [
      {
        "deleted": false,
        "tag": false,
        "branch": true,
        "new": true,
        "alreadyUpdated": false,
        "local": "refs/heads/renovate/pin-dependencies",
        "remote": "refs/heads/renovate/pin-dependencies"
      }
    ],
    "branch": {
      "local": "renovate/pin-dependencies",
      "remote": "renovate/pin-dependencies",
      "remoteName": "origin"
    },
    "remoteMessages": {
      "all": [
        "Create a pull request for 'renovate/pin-dependencies' on GitHub by visiting:",
        "https://github.com/AaronBuxbaum/renovate-repro/pull/new/renovate/pin-dependencies"
      ],
      "pullRequestUrl": "https://github.com/AaronBuxbaum/renovate-repro/pull/new/renovate/pin-dependencies"
    }
  }
}
INFO: Branch created(branch="renovate/pin-dependencies")
{
  "commitSha": "a946966"
}
DEBUG: Ensuring PR(branch="renovate/pin-dependencies")
DEBUG: There are 0 errors and 0 warnings(branch="renovate/pin-dependencies")
DEBUG: getBranchPr(renovate/pin-dependencies)(branch="renovate/pin-dependencies")
DEBUG: findPr(renovate/pin-dependencies, undefined, open)(branch="renovate/pin-dependencies")
DEBUG: findPr(renovate/pin-dependencies, undefined, closed)(branch="renovate/pin-dependencies")
DEBUG: Creating PR(branch="renovate/pin-dependencies")
{
  "prTitle": "Pin dependency lodash to 4.17.21"
}
DEBUG: Creating PR(branch="renovate/pin-dependencies")
{
  "title": "Pin dependency lodash to 4.17.21",
  "head": "AaronBuxbaum:renovate/pin-dependencies",
  "base": "master",
  "draft": false
}
DEBUG: PR created(branch="renovate/pin-dependencies")
{
  "pr": 3,
  "draft": false
}
DEBUG: Adding labels '' to #3(branch="renovate/pin-dependencies")
INFO: PR created(branch="renovate/pin-dependencies")
{
  "pr": 3,
  "prTitle": "Pin dependency lodash to 4.17.21"
}
DEBUG: Created Pull Request #3(branch="renovate/pin-dependencies")
DEBUG: Checking #3 for automerge(branch="renovate/pin-dependencies")
{
  "automerge": false,
  "automergeType": "pr",
  "automergeComment": "automergeComment"
}
DEBUG: No automerge(branch="renovate/pin-dependencies")
DEBUG: getBranchPr(renovate/pin-dependencies)
DEBUG: findPr(renovate/pin-dependencies, undefined, open)
DEBUG: Found PR #3
DEBUG: Returning from graphql open PR list
DEBUG: Removing any stale branches
DEBUG: config.repoIsOnboarded=true
DEBUG: Branch lists
{
  "branchList": [
    "renovate/pin-dependencies"
  ],
  "renovateBranches": [
    "renovate/pin-dependencies"
  ]
}
DEBUG: remainingBranches=
DEBUG: No branches to clean up
DEBUG: Repository timing splits (milliseconds)
{
  "splits": {
    "init": 2949,
    "extract": 1400,
    "lookup": 130,
    "update": 18050
  },
  "total": 22937
}
DEBUG: http statistics
{
  "hostStats": [
    "api.github.com, 7 requests, 307ms request average, 0ms queue average"
  ],
  "totalRequests": 7
}
INFO: Repository finished
{
  "durationMs": 22937
}
```

**Have you created a minimal reproduction repository?**
https://github.com/AaronBuxbaum/renovate-repro

Note the issue here: https://github.com/AaronBuxbaum/renovate-repro/pull/3/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519

- [x] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

### PR DESCRIPTION
**Pull request by PhilipAbed**
## Changes
added support to yarn metadata dependency version, it was ignored.

## Context

Closes #9516

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [X] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/npm/update/dependency/index.ts [hunk 1]
```diff
@@ -20,6 +21,15 @@ function replaceAsString(
...
       [newValue]: parsedContents[depType][oldValue],
     })[oldValue];
+  } else if (depType === 'dependenciesMeta') {
+    if (oldValue !== newValue) {
+      Object.defineProperty(
+        parsedContents[depType],
+        newValue,
+        Object.getOwnPropertyDescriptor(parsedContents[depType], oldValue)
+      );
+      delete parsedContents[depType][oldValue];
```
Comment (viceice) [inline lib/modules/manager/npm/update/dependency/index.ts:31]: Does this change the order? please validate to not change the key order.

**Comment by PhilipAbed** [comment]
@rarkins ofcourse, this is the same project from the same guy from the last task (Yarn patch clobbered) 
it was same project, same man, same test case, just 2 different issues, 
1 for Patch protocol being updated wrongly
1 for dependencies metadata is ignored.

i waited until the Yarn patch clobbered is Merged, so i can finish this up.

**Comment by PhilipAbed** [comment]
i will re-run again to be sure its still working with latest changes (even though test works and it checks it)

**Comment by PhilipAbed** [comment]
@rarkins  here's the run: 
https://github.com/StinkyLord/renovate-repro-9509/pull/5/files

File: lib/modules/manager/npm/update/dependency/index.ts [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
+  oldObj: DependenciesMeta,
+  oldKey: string,
+  newKey: string
+): DependenciesMeta {
+  const keys = Object.keys(oldObj);
+  return keys.reduce((acc, val) => {
+    if (val === oldKey) {
+      acc[newKey] = oldObj[oldKey];
+    } else {
+      acc[val] = oldObj[val];
```
Comment (viceice) [inline lib/modules/manager/npm/update/dependency/index.ts:18]:
```suggestion
  return keys.reduce((acc, key) => {
    if (key=== oldKey) {
      acc[newKey] = oldObj[oldKey];
    } else {
      acc[key] = oldObj[key];
```

So it's more clear that it's the property name and not the  value.

**Review by viceice — DISMISSED** [review]

**Comment by PhilipAbed** [comment]
fixed lint issue

**Review by viceice — APPROVED** [review]

**Pull request merged**